x-pack/plugins/ml/common/util/date_utils.ts

```diff
@@ -31,6 +31,17 @@ export function validateTimeRange(time?: TimeRange): boolean {
   return !!(momentDateFrom && momentDateFrom.isValid() && momentDateTo && momentDateTo.isValid());
 }
 
+export function createAbsoluteTimeRange(time: TimeRange) {
+  if (validateTimeRange(time) === false) {
+    return null;
+  }
+
+  return {
+    to: dateMath.parse(time.to)?.valueOf(),
```

Comment:
This is not accurate - the to part has to be rounded up while the from part is rounded down. Please use `data.query.timefilter.timefilter.calculateBounds` instead

Reply:
Fixed in https://github.com/elastic/kibana/pull/129762/commits/3ba9b7cc140748fb6916d50eb94c96f66ef90490